pkg/operator/sync.go

```diff
@@ -336,15 +336,10 @@ func (optr *Operator) checkMinimumWorkerMachines() error {
 		nonRunningMachines = append(nonRunningMachines, nonRunningMachineSetMachines...)
 	}
 
-	if expectedReplicas == 0 {
-		// This means there are no MachineSets in the cluster, so we are ok to proceed.
-		return nil
-	}
-
 	// If any MachineSet doesn't have the correct number of replicas, we error before this point.
 	// So the running replicas should be (total replicas) - (non-running replicas).
 	runningReplicas := expectedReplicas - int32(len(nonRunningMachines))
-	if runningReplicas < minimumWorkerReplicas {
+	if runningReplicas < expectedReplicas && runningReplicas < minimumWorkerReplicas {
```

Comment:
What do you think about removing the check on `minimumWorkerReplicas` entirely? It shouldn't be needed and it's not even enforced by the openshift-installer itself. The `minimumWorkerReplicas` is used by the installer to determine if the masters should be schedulable and what Infrastructure topology to use (maybe some other places too) but it's not used to blcok/error the installation. Hardcoding this number here may cause more issues in the future.

Reply:
I think it is still needed, in the case of a 3 machine machineset, if 2 are running and 1 is not, then `runningReplicas < expectedReplicas` is true but `runningReplicas < minimumWorkerReplicas` is false, so  we don't error out here, which is desired

I think this makes sense, if the machineset is at total capacity, then of course that's not an error state, so this should solve the issue of a single machine machineset